typify-impl/src/test_util.rs

```diff
@@ -81,6 +88,10 @@ fn validate_output_impl<T: JsonSchema + Schema>(ignore_variant_names: bool) {
     assert!(!file.items.is_empty(), "{}", output.to_string());
     let actual = parse2::<DeriveInput>(file.items.first().unwrap().to_token_stream()).unwrap();
 
+    // Decanonicalize the types generated by typify so that we can compare them to the original
+    // Rust types' ASTs that definitely do not use canonicalized std types.
+    let actual = decanonicalize_std_types(actual);
+
     // Make sure they match.
     if let Err(err) = expected.syn_cmp(&actual, ignore_variant_names) {
```

Comment:
would another approach be to change the implementation of `impl SynCompare for TypePath` to be fuzzier i.e. to identify `::std::...::Foo` as equivalent to `Foo`?

Alternatively, I'm inclined towards changing the tests to just use the fully qualified type name.

What do you think of these options?

Reply:
> Yes I suppose you could change the `SynCompare` impl to strip path components any time it sees `::std::` before the comparison. My personal preference is to not do that, since it's more spooky action at a distance, while this impl is very explicit about what's happening.

Fair enough.

> That would certainly work, but if anyone adds new test cases will they remember that the types have to be fully-qualified, and if they forget will the resulting test failure make it obvious what their mistake was?

I think so.

> It's your codebase, your call, but speaking in terms of my personal preference, I prefer the explicit decanonicalization approach.

Let's go with your approach.